packages/react-menu/src/components/MenuItem/MenuItem.tsx

```diff
@@ -12,6 +14,7 @@ export const MenuItem = React.forwardRef<HTMLElement, MenuItemProps>((props, ref
   const state = useMenuItem(props, ref);
 
   useMenuItemStyles(state);
+  useCheckmarkStyles((state as unknown) as MenuItemCheckboxState);
```

Comment:
Perhaps we could make `MenuItemState` compatible with `MenuItemCheckboxState` to avoid this

Reply:
yeah some TS fu can be done here, but my main requirement is that

> This slot should only be used as an alignment helper, if you want to actually apply meaning to it then use MenuItemCheckbox and MenuItemRadio

There is a reason those components that `name`, `value` and `checked` props too